File: livebox/src/main/java/com/creations/livebox/datasources/LocalDataSource.java
package com.creations.livebox.datasources;

import com.creations.livebox.util.Optional;

/**
 * @author Sérgio Serra on 25/08/2018.
 * Criations
 * sergioserra99@gmail.com
 */
public interface LocalDataSource<I> {
    Optional<I> read(String key);

    void save(String key, I input);

    void clear(String key);
}

File: livebox/src/main/java/com/creations/livebox/datasources/LocalDataSource.kt
package com.creations.livebox.datasources

import java.lang.reflect.Type

/**
 * @author Sérgio Serra on 25/08/2018.
 * sergioserra99@gmail.com
 */
interface LocalDataSource<I, T> {
    val type: Type
    fun read(key: String): T?
    fun save(key: String, input: I)
    fun clear(key: String)
}
